suprb/optimizer/solution/ga/base.py

```diff
@@ -78,30 +110,291 @@ def _optimize(self, X: np.ndarray, y: np.ndarray):
+        If False, solutions are generated new for every `optimize()` call.
+        If True, solutions are used from previous runs.
+    n_jobs: int
+        The number of threads / processes the optimization uses.
+    """
+
+    n_elitists_: int
+    fitness_variance_min: float
+    fitness_variance_max: float
+    old_gen: list[Solution]
+    fitness_mean: float
+    fitness_min: float
+    fitness_max: float
+
+    def __init__(
+        self,
+        n_iter: int = 32,
+        population_size: int = 32,
+        elitist_ratio: float = 0.17,
+        mutation: SolutionMutation = BitFlips(),
+        crossover: SolutionCrossover = NPoint(n=3),
+        selection: SolutionSelection = Tournament(),
+        mutation_rate_min: float = 0.01,
+        mutation_rate_max: float = 0.1,
+        crossover_rate_min: float = 0.4,
+        crossover_rate_max: float = 0.9,
+        init: SolutionInit = RandomInit(),
+        archive: SolutionArchive = Elitist(),
+        random_state: int = None,
+        n_jobs: int = 1,
+        warm_start: bool = True,
+    ):
+        super().__init__(
+            n_iter=n_iter,
+            population_size=population_size,
+            init=init,
+            archive=archive,
+            random_state=random_state,
+            n_jobs=n_jobs,
+            warm_start=warm_start,
+        )
+
+        self.mutation_rate_min = mutation_rate_min
+        self.mutation_rate_max = mutation_rate_max
+        self.mutation_rate_current_max = mutation_rate_max
+        self.crossover_rate_min = crossover_rate_min
+        self.crossover_rate_current_min = crossover_rate_min
+        self.crossover_rate_max = crossover_rate_max
+        self.mutation = mutation
+        self.crossover = crossover
+        self.selection = selection
+        self.elitist_ratio = elitist_ratio
+
+    def fitness_calculation(self):
+        fitness_variance = np.var([i.fitness_ for i in self.population_])
+        self.fitness_variance_min = fitness_variance
+        self.fitness_variance_max = fitness_variance
+        self.old_gen = self.population_
+        self.fitness_mean = np.mean([i.fitness_ for i in self.population_])
+        self.fitness_min = np.min([i.fitness_ for i in self.population_])
+        self.fitness_max = np.max([i.fitness_ for i in self.population_])
+
+    def crossover_children(self, parent_pairs):
+        return list(
+            flatten(
+                [
+                    (
+                        self.crossover.saga2_crossover(
+                            A,
+                            B,
+                            self.crossover_rate_current_min,
+                            self.crossover_rate_max,
+                            self.fitness_mean,
+                            self.fitness_min,
+                            self.fitness_max,
+                            random_state=self.random_state_,
+                        ),
+                        self.crossover.saga2_crossover(
```

Comment:
currently, our crossover methods compute (but don't return) both children. We should use this here

Reply:
If I am not mistaken, this is the crossover method in question: 
![grafik](https://github.com/user-attachments/assets/0ede4351-9880-4a4c-82a5-ed8b1838ad28)

Where we only return one